common/meta/grpc/src/meta_client_on_kv.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::convert::Infallible;
+use std::fmt::Debug;
+use std::string::FromUtf8Error;
+use std::time::Duration;
+
+use common_base::escape_for_key;
+use common_exception::ErrorCode;
+use common_grpc::RpcClientTlsConfig;
+use common_meta_types::DatabaseIdent;
+use common_meta_types::DatabaseNameIdent;
+use common_meta_types::DatabaseTenantIdIdent;
+use common_tracing::tracing;
+
+use crate::MetaGrpcClient;
+use crate::MetaGrpcClientConf;
+
+/// Metasrv client base on only KVApi.
+///
+/// It implement transactional operation on the client side, through `KVApi::transaction()`.
+///
+/// TODO(xp): consider remove the `impl MetaAPI for MetaGrpcClient` when this is done.
+pub struct MetaClientOnKV {
+    pub(crate) inner: MetaGrpcClient,
+}
+
+impl MetaClientOnKV {
+    pub async fn try_new(conf: &MetaGrpcClientConf) -> Result<MetaClientOnKV, Infallible> {
+        Ok(Self {
+            inner: MetaGrpcClient::try_new(conf).await?,
+        })
+    }
+
+    #[tracing::instrument(level = "debug", skip(password))]
+    pub async fn try_create(
+        addr: &str,
+        username: &str,
+        password: &str,
+        timeout: Option<Duration>,
+        conf: Option<RpcClientTlsConfig>,
+    ) -> Result<Self, ErrorCode> {
+        Ok(Self {
+            inner: MetaGrpcClient::try_create(addr, username, password, timeout, conf).await?,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct DatabaseIdGen {}
+
+#[derive(Debug)]
+pub struct TableIdGen {}
+
+/// Convert to a string as a key used in KVApi
+pub trait ToKVMetaKey: Debug {
+    fn to_key(&self) -> Result<String, FromUtf8Error>;
+}
+
+impl ToKVMetaKey for DatabaseNameIdent {
+    fn to_key(&self) -> Result<String, FromUtf8Error> {
+        let k = format!(
+            "__fd_database/{}/by_name/{}",
```

Comment:
prefix like "__fd_database" should define as CONST with a name if prefered.

Reply:
It feels a bit weird by using a const.
`impl ToKVMetaKey for DatabaseNameIden` should be the only place using this literal string.

I'd keep this, unless there is a second place using this literal.